test/langtools/tools/javac/platform/createsymbols/CreateSymbolsTest.java

```diff
@@ -80,13 +85,37 @@ void doRun() throws Exception {
         JavaCompiler compiler = ToolProvider.getSystemJavaCompiler();
 
         try (StandardJavaFileManager fm = compiler.getStandardFileManager(null, null, null)) {
-            compiler.getTask(null,
-                             null,
-                             null,
-                             Arrays.asList("-d", compileDir.toAbsolutePath().toString()),
-                             null,
-                             fm.getJavaFileObjects(createSymbols, createTestImpl, toolBox)
-                            ).call();
+            List<Path> files = new ArrayList<>();
+
+            files.add(createSymbols);
+            files.add(createTestImpl);
+
+            Files.list(toolBox)
+                 .forEach(files::add);
+
+            Boolean res =
+                    compiler.getTask(null,
+                                      null,
+                                      null,
+                                      List.of("-d",
+                                              compileDir.toAbsolutePath().toString(),
+                                              "-g",
+                                              "--add-modules", "jdk.jdeps",
+                                              "--add-exports", "jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED",
+                                              "--add-exports", "jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED",
+                                              "--add-exports", "jdk.compiler/com.sun.tools.javac.jvm=ALL-UNNAMED",
+                                              "--add-exports", "jdk.compiler/com.sun.tools.javac.main=ALL-UNNAMED",
+                                              "--add-exports", "jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED",
+                                              "--add-exports", "jdk.javadoc/jdk.javadoc.internal.api=ALL-UNNAMED",
+                                              "--add-exports", "jdk.javadoc/jdk.javadoc.internal.tool=ALL-UNNAMED",
+                                              "--add-exports", "jdk.jdeps/com.sun.tools.classfile=ALL-UNNAMED",
+                                              "--add-exports", "jdk.jdeps/com.sun.tools.javap=ALL-UNNAMED"),
+                                      null,
+                                      fm.getJavaFileObjectsFromPaths(files)
+                                    ).call();
+            if (!res) {
+                throw new IllegalStateException("Cannot compile test.");
+            }
         }
 
         URLClassLoader cl = new URLClassLoader(new URL[] {testClasses.toUri().toURL(), compileDir.toUri().toURL()});
```

Comment:
line 132: the name of the local variable `createSymbols` seems more specific than the method might imply!